### PR TITLE
Added metrics and health endpoints

### PR DESCRIPTION
**Pull request by ronniel1**
Metrics endpoint dumps the basic service metrics in Prometheus format

### CODE REVIEW

**Comment by ronniel1** [comment]
> Nice! Do we need it to be under /api/assisted-install/v1/ for SRE guidelines?

I don't know about SRE guidelines, but for [openshift monitoring conventions](https://kubernetesbyexample.com/healthz/) it does not

File: cmd/main.go [hunk 1]
```diff
@@ -121,3 +122,13 @@ func withMetricsMiddleware(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r)
 	})
 }
+
+func withHealthMiddleware(next http.Handler) http.Handler {
```
Comment (ori-amizur) [inline cmd/main.go:126]: Why not add new handler with http.HandleFunc or http.Handle ?

Comment (ronniel1) [reply cmd/main.go:126]: What is wrong with the middleware approach?
Adding a new handler will require another port

Comment (ori-amizur) [reply cmd/main.go:126]: It is more in line with the http library approach.  You let the http library do the dispatching.
Adding new handler does not require a new port

File: cmd/main.go [hunk 2]
```diff
@@ -93,10 +96,39 @@ func main() {
...
+
 	h = requestid.Middleware(h)
 	if err != nil {
 		log.Fatal("Failed to init rest handler,", err)
 	}
 
 	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", swag.StringValue(port)), h))
 }
+
+func withMetricsMiddleware(next http.Handler) http.Handler {
```
Comment (filanov) [inline cmd/main.go:116]: suggesting to move this to /pkg
lets have a clean main.

File: cmd/main.go [hunk 3]
```diff
@@ -93,10 +96,39 @@ func main() {
...
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodGet && r.URL.Path == "/metrics" {
+			promhttp.Handler().ServeHTTP(w, r)
+			return
+		}
+		next.ServeHTTP(w, r)
+	})
+}
+
+func withHealthMiddleware(next http.Handler) http.Handler {
```
Comment (filanov) [inline cmd/main.go:126]: suggesting to move this to /pkg
lets have a clean main.

File: cmd/main.go (hunk 1 above)
Comment (ronniel1) [reply cmd/main.go:126]: I don't agree that is is more inline.
Adding a handler is one acceptable way, and doing it this way is another
This has the benefit of not requiring another port

Comment (filanov) [reply cmd/main.go:126]: i like the middleware solution this way we can add more functionality if needed.

File: cmd/main.go (hunk 3 above)
Comment (ronniel1) [reply cmd/main.go]: done

File: cmd/main.go (hunk 2 above)
Comment (ronniel1) [reply cmd/main.go]: done

**Pull request merged**